### ISSUE TITLE
LiteralInInterpolation false positive when interpolating "#{__LINE__}"

### ISSUE THREAD

**Issue by rrosenblum** [bbatsov/rubocop#1769]
I just upgraded to RuboCop 0.30.0 and ran into a false positive when running LiteralInInterpolation.

Here is a test that exercises the bug.
```ruby
it 'accepts strings like __LINE__' do
  inspect_source(cop, '"this is #{__LINE__} silly"')
  expect(cop.offenses).to be_empty
end


1) RuboCop::Cop::Lint::LiteralInInterpolation accepts strings like __LINE__
     Failure/Error: expect(cop.offenses).to be_empty
       expected `[W:  1: 12: Literal interpolation detected.].empty?` to return true, got false
```

I will try to find a fix for the issue, but in the mean time, I wanted to bring exposure to the bug.

**Comment by rrosenblum** [comment]
I am not sure of the proper way to fix this. After looking into the code, the error is caused by `"#{__LINE__}"` having the type `":int` and `"#{__FILE__}"` having the type `:str`. When I update the method `special_string?` to account for `:int`, the tests for `"#{1}"` begin to fail. I cannot figure out a way to distinguish between `"#{__LINE__}"` and `"#{1}"`.

**Comment by jonas054** [comment]
I think you must look at the actual source to sort out this special case.
```ruby
node.loc.expression.is?('__LINE__')
```

**Comment by bbatsov** [comment]
Yeah, that should solve it. Unfortunately its a somewhat expensive check, but I can think of no other alternatives.
